### DISCUSSION TITLE
How to get all my collected movies/tvshows?

### DISCUSSION THREAD

**Discussion by fatshotty** [trakt/trakt-api#321] [Q&A] [answered]
Hello there,
I would create my own app based on trakt.tv. I need to show all my collected movies and tvshows in the homepage of my app, but I cannot find the correct API.
Is there a way to get all my collected movies/tvshows? Am I in wrong?

thanks in advance

**Comment by henrikfroehling** [comment]
Hi, I think you're looking for this API endpoint: https://trakt.docs.apiary.io/#reference/users/collection/get-collection

**Reply by fatshotty** [reply]
thanks for reply.
Yep, it seems to be the endpoint I was looking for :D
However, I'm using the test client apiary web site and the endpoint correctly returns 200 with entire movies list. It is with the example name "sean". If I use my username-slug it returns 401 :/
I correctly set my client_id as trakt-api-key, but it returns 401 using my username-slug

This is the example code that works fine:
```
request({
  method: 'GET',
  url: 'https://api.trakt.tv/users/sean/collection/movies',
  headers: {
    'Content-Type': 'application/json',
    'trakt-api-version': '2',
    'trakt-api-key': 'XXXXXXX'
  }}, function (error, response, body) {
  console.log('Status:', response.statusCode);
  console.log('Headers:', JSON.stringify(response.headers));
  console.log('Response:', body);
});
```

If I use my username-slug , it returns 401


Any help?

**Reply by henrikfroehling** [reply]
I'm currently not sure why this API endpoint is stated as "OAuth optional", because it's not very clear to me when an OAuth authorization is required and when not.

But status code (401) states that you're not authorized. I tested the API endpoint the same way you did and got the same result. Works with username "sean" but not with my own.

If get authenticated (like in https://trakt.docs.apiary.io/#reference/authentication-oauth or https://trakt.docs.apiary.io/#reference/authentication-devices), it works.

FYI: I did the requests with Postman.

**Reply by fatshotty** [reply]
thanks for reply
I got authenticated via "auth-device" and always get 401.
Now I re-create a new app with new client_id and client_secret; made a new auth-device and get new access_token.
Now the endpoint works.

I don't know why I need to create a new app... but now it works again :/

Thanks again

**Reply by henrikfroehling** [reply]
I'm just guessing here: Was your access token possibly outdated?
Usually, if there is not an API error, you only need to refresh the access token every three months.
Refreshing an access token works with the following API endpoint: https://trakt.docs.apiary.io/#reference/authentication-oauth/get-token/exchange-refresh_token-for-access_token

**Reply by fatshotty** [reply]
thanks
already tried to re-new my access token but 401 was still there :/

**Comment by rectifyer** [comment, marked as answer]
> OAuth optional

If the user account is private, OAuth is required. If the user account is public, no OAuth is required. It’s always the best idea to send valid OAuth since that also allows you to view any of your friends with private data.

**Reply by henrikfroehling** [reply]
Thanks for the clarification.

But there still seems to be an error. I tested following scenario.
1. Authenticate -> valid access token
2. Get my user settings ("users/settings") indicating that my profile is marked as public
![TraktUserSettings](https://user-images.githubusercontent.com/3494788/147283633-2e74f4ab-f300-46f8-bb53-3850fcf533fc.png)
3. Calling with my username "users/ixxus"collection/movies" with status code 200
4. Revoke my access token
5. Calling with my username "users/ixxus/collection/movies" with status code 401

In the last step I would assume the status code 200 for "users/ixxus/collection/movies", since my profile is public and authorization is optional.

**Reply by rectifyer** [reply]
If you send a OAuth access token, we're still going to check it and make sure it's valid. If it's invalid, it will error out.
